File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/GetMovieUseCase.java
package com.hwonchul.movie.domain.usecase.details;

import com.hwonchul.movie.domain.model.MovieDetail;

import io.reactivex.rxjava3.core.Flowable;

/**
 * 영화상세 item 가져오기 Usecase
 */
public interface GetMovieUseCase {

    Flowable<MovieDetail> invoke(int movieId);
}

File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/GetMovieUseCase.kt
package com.hwonchul.movie.domain.usecase.details

import com.hwonchul.movie.domain.model.MovieDetail
import kotlinx.coroutines.flow.Flow

/**
 * 영화상세 item 가져오기 Usecase
 */
interface GetMovieUseCase {

    suspend operator fun invoke(movieId: Int): Flow<Result<MovieDetail>>
}